src/core/symbol.c

```diff
@@ -38,190 +39,143 @@
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_NS_OPTION_TYPE, "NN_NS_OPTION_TYPE", NN_NS_NAMESPACE,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-	 {NN_NS_OPTION_UNIT, "NN_NS_OPTION_UNIT", NN_NS_NAMESPACE,
-			NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_NS_FLAG, "NN_NS_FLAG", NN_NS_NAMESPACE,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_NS_ERROR, "NN_NS_ERROR", NN_NS_NAMESPACE,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_NS_LIMIT, "NN_NS_LIMIT", NN_NS_NAMESPACE,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_NS_EVENT, "NN_NS_EVENT", NN_NS_NAMESPACE,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-
-    {NN_TYPE_NONE, "NN_TYPE_NONE", NN_NS_OPTION_TYPE,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_TYPE_INT, "NN_TYPE_INT", NN_NS_OPTION_TYPE,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_TYPE_STR, "NN_TYPE_STR", NN_NS_OPTION_TYPE,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-
-    {NN_UNIT_NONE, "NN_UNIT_NONE", NN_NS_OPTION_UNIT,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_UNIT_BYTES, "NN_UNIT_BYTES", NN_NS_OPTION_UNIT,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_UNIT_MILLISECONDS, "NN_UNIT_MILLISECONDS", NN_NS_OPTION_UNIT,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_UNIT_PRIORITY, "NN_UNIT_PRIORITY", NN_NS_OPTION_UNIT,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_UNIT_BOOLEAN, "NN_UNIT_BOOLEAN", NN_NS_OPTION_UNIT,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-
-    {NN_VERSION_CURRENT, "NN_VERSION_CURRENT", NN_NS_VERSION,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_VERSION_REVISION, "NN_VERSION_REVISION", NN_NS_VERSION,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_VERSION_AGE, "NN_VERSION_AGE", NN_NS_VERSION,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-
-    {AF_SP, "AF_SP", NN_NS_DOMAIN,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {AF_SP_RAW, "AF_SP_RAW", NN_NS_DOMAIN,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-
-    {NN_INPROC, "NN_INPROC", NN_NS_TRANSPORT,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_IPC, "NN_IPC", NN_NS_TRANSPORT,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_TCP, "NN_TCP", NN_NS_TRANSPORT,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_WS, "NN_WS", NN_NS_TRANSPORT,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-
-    {NN_PAIR, "NN_PAIR", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_PUB, "NN_PUB", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_SUB, "NN_SUB", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_REP, "NN_REP", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_REQ, "NN_REQ", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_PUSH, "NN_PUSH", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_PULL, "NN_PULL", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_SURVEYOR, "NN_SURVEYOR", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_RESPONDENT, "NN_RESPONDENT", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
-    {NN_BUS, "NN_BUS", NN_NS_PROTOCOL,
-        NN_TYPE_NONE, NN_UNIT_NONE},
+#define	NN_SYM(sym, namespace, typ, unit) \
```

Comment:
Yeah, that's a formatting bug in github. Go figure.

There's a real benefit here, which is that the stringifcation is now automated, eliminating a rather large opportunity for human error.  (The symbol names will fail a compile check if they are wrong. :-)

The compact representation makes it easier to see what's going on.  I understand your gripe about breaking the IDE search function, but I think the benefits outweigh the cons in this case.  (In fact, you'll still find the reference nearby in this file, because the symbols that are "abbreviated" are listed in full in their own entries in this array.)